src/pages/SchoolsMapPage.tsx

```diff
@@ -118,38 +120,52 @@ const SchoolsMapPage: FC<RouteComponentProps> = () => {
   const filters = useFilters(filtersDefinition, deserializeFilters(p, filtersDefinition));
   const [notListedCount, setNotListedCount] = useState(0);
   const [notListedVisible, setNotListedVisible] = useState(true);
-  const searchData = {
-    query,
-    ...convertFilterStateToObject(filters.filtersState),
-  };
-  const { data: schools, error } = useAllSchools(searchData);
+  const [getSchools, { data, error }] = useSchoolsMapData();
+
+  const filtersObj = convertFilterStateToObject(filters.filtersState);
+
+  const delayedGetSchools = useCallback(debounce(getSchools, 300), [getSchools]);
 
   useDeepCompareEffect(() => {
+    const searchData = {
+      query,
+      ...filtersObj,
+    };
     currUrl.search = serializeSearchData(searchData, 'search');
     window.history.replaceState(null, '', currUrl.toString());
-  }, [searchData, currUrl]);
+    delayedGetSchools(query, filters.filtersState);
+  }, [query, filtersObj]);
 
-  const bounds = useMemo(() => {
-    let b = [];
+  const schools = data?.allSchools;
+
+  const bounds: LatLngBoundsLiteral | undefined = useMemo(() => {
+    let b: LatLngBoundsLiteral = [];
     setNotListedCount(0);
     setNotListedVisible(true);
-    if (schools && schools.length > 0) {
-      b = schools.reduce((prev: any, school: any) => {
-        if (!doesSchoolHaveCoords(school)) {
-          setNotListedCount((count) => count + 1);
-          return prev;
-        }
-
-        return [...prev, getSchoolCoords(school) as LatLngExpression];
-      }, []);
+    if (schools && schools.edges.length > 0) {
+      b = (schools.edges as { node: ISchoolNode }[]).reduce<LatLngBoundsLiteral>(
```

Comment:
Why are there so many type casts throughout this change?

Reply:
We use graphql codegen to generate types so we need to cast them sometimes. I think we can avoid doing so but for now maybe we can just let's leave it as it is :stuck_out_tongue: ?